### PR TITLE
Users/suaggar/function on linux

### PR DESCRIPTION
**Pull request by SumiranAgg**
Not ready for review. Pending: 
- Runtime stack for function Linux
  Adding static picklist
- Difference between function on Linux and container
  they both are identical, containers and code are different flavours which require different deployment 
  methods

Includes few bug fixes:
1. in beginRequest we now allow retry if reqOptions is undefined.
2. WebRequestOptions object was missing in some places where reqOptions were passed as an argument.
3. Whats new in version 4.* text update

### CODE REVIEW

File: Tasks/Common/azure-arm-rest/azure-arm-app-service-kudu.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export class KuduServiceManagementClient {
                         tl.warning(tl.loc('ASE_SSLIssueRecommendation'));
                 }
 
-                if(retryCount > 0 && exceptionString.indexOf('Request timeout') != -1 && reqOptions && reqOptions.retryRequestTimedout) {
+                if(retryCount > 0 && exceptionString.indexOf('Request timeout') != -1 && (!reqOptions || reqOptions.retryRequestTimedout)) {
```
Comment (kmkumaran) [inline Tasks/Common/azure-arm-rest/azure-arm-app-service-kudu.ts]: Why this change?

File: Tasks/AzureRmWebAppDeploymentV4/task.json [hunk 2]
```diff
@@ -18,22 +18,22 @@
     "version": {
         "Major": 4,
         "Minor": 3,
-        "Patch": 7
+        "Patch": 8
     },
-    "releaseNotes": "What's new in version 4.* (preview)<br />Supports Kudu Zip Deploy<br />Supports App Service Environments<br />Improved UI for discovering different App service types supported by the task<br/>Click [here](https://aka.ms/azurermwebdeployreadme) for more information.",
+    "releaseNotes": "What's new in version 4.* (preview)<br />Supports Zip Deploy, Run From Zip, War Deploy<br />Supports App Service Environments<br />Improved UI for discovering different App service types supported by the task<br/>Run From Zip is the preferred deployment method, which makes files in wwwroot folder read-only<br/>Click [here](https://aka.ms/azurermwebdeployreadme) for more information.",
```
[the remark this answers is not in the document]
Comment (SumiranAgg) [reply Tasks/AzureRmWebAppDeploymentV4/task.json]: Will take it up in next PR
